src/operations/SymbolicOperation.cpp

```diff
@@ -332,4 +332,54 @@ void SymbolicOperation::instantiate(const VariableAssignment& assignment) {
   }
   checkUgate();
 }
+
+void SymbolicOperation::invert() {
+  auto negate = [this](const size_t index) {
+    if (isSymbolicParameter(index)) {
+      symbolicParameter.at(index) = -symbolicParameter.at(index).value();
+    } else {
+      parameter.at(index) = -parameter.at(index);
+    }
+  };
+  auto add = [this](const size_t index, const fp value) {
+    if (isSymbolicParameter(index)) {
+      symbolicParameter.at(index) = symbolicParameter.at(index).value() + value;
+    } else {
+      parameter.at(index) += value;
+    }
+  };
```

Comment:
no real need to define these as lambdas here. Simply declare these as private methods of the `SymbolicOperation` class (probably using a slightly more descriptive name).